### ISSUE TITLE
Need for Speed 2016

### ISSUE THREAD

**Issue by EmSuperCraft** [x360ce/x360ce#522]
Anybody know why Need for Speed crashes in main menu whan you follow these instructions: https://www.youtube.com/watch?v=7XG0duaMyPs
I am not the only one having this issue.

**Comment by bugz8unny69** [comment]
It's been awhile, have you been able to fixed this issue without needing to rename asiloader.dll to dinput_x64.dll? (I haven't tested, but with stock configuration, the game would crash on start menu)

**Comment by EmSuperCraft** [comment]
No, haven't been able without it. Bought an xinput controller now XD. But still use my dinput controllers for splitscreen games or multiplayer stuff like trine.
With stock it used to crash and then a while it worked while I had it renamed, but crashed at some point in the game and only would start again when I renamed the .dll. But sometime also that didn't work anymore. Maybe the team just has to make a patch or so.
(btw when it worked some controls were somehow layered, if that is the right word. but I couldn't find any xinput files in the program without x360ce.)

**Comment by bugz8unny69** [comment]
I actually opted for InputMapper2 with DirectInput, SCP 360 device emulator, and Mapping plugin with my DirectInput controller and NFS recognizes it as XInput 360 controller. Additionally, it avoids the cumbersome of copying over files and setting up controller settings for every game. As far as some controls are layered issue, that's a Need For Speed issue, Xbox 360 controller is not a supported controller and it's recognize as a controller and a wheel. One would need to unmap the controls for the Wheel.

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
